sahi/model.py

```diff
@@ -92,6 +92,12 @@ def set_model(self, model: Any, **kwargs):
         """
         raise NotImplementedError()
 
+    def set_device(self):
+        """
+        Sets the device for the model.
+        """
+        raise NotImplementedError()
```

Comment:
not implemented şekilde değil de torch device şeklinde olacak bu fonksiyon:

# automatically set device if its None
        if not (self.device):
            self.device = "cuda:0" if is_torch_cuda_available() else "cpu"

Reply:
Neden torch ekliyoruz anlamadım. Eğer set_device kullanılmamışsa torch kütüphanesi aktif olacak şekilde kodladım. Bunu demek ıstedınız galiba.

Reply:
base class icerisinde torch set device olacak, cunku kullandigimiz modellerin cogu torch uzerine. eger kullanici yeni bir class olustururuken tensorflow ya da paddle kullanmak istiyorsa set_device fonksiyonunu owerwrite ederek baska bir device destegi ekleyebilir. biz de tensorflowhub classinda bunu yapiyoruz.